projects/observability/src/shared/components/topology/d3/layouts/graph-layout.ts

```diff
@@ -0,0 +1,76 @@
+import { RenderableTopology, RenderableTopologyNode, TopologyCoordinates, TopologyEdge, TopologyNode } from "../../topology";
+
+export class GraphLayout {
+
+  private levelToNodesMap = new Map<number, RenderableTopologyNode[]>();
+  private nodeToLevelMap = new Map<RenderableTopologyNode, number>();
+  private nodeCoOrdinatesMap: Map<TopologyNode, TopologyCoordinates> = new Map<TopologyNode, TopologyCoordinates>();
+
+  public layout(topology: RenderableTopology<TopologyNode, TopologyEdge>): void {
+
+    this.findRootNodes(topology);
+    this.fillNodeAndLevelMaps();
+    this.fillNodeCoOrdinatesMap();
```

Comment:
please fix capitalization - Coordinates is one word

Reply:
updated